client/lib/query-args/index.ts

```diff
@@ -0,0 +1,59 @@
+import { getQueryArgs } from '@wordpress/url';
+import page from 'page';
+
+function getRelativeUrlWithParameters( queryArgs: object ): string {
+	const urlParams = new URLSearchParams( window.location.search );
+	let returnPath = window.location.pathname;
+
+	if ( queryArgs ) {
+		for ( const [ key, value ] of Object.entries( queryArgs ) ) {
+			if ( '' !== value ) {
+				urlParams.set( key, value );
+			} else {
+				urlParams.delete( key );
+			}
+		}
+
+		returnPath += '?' + urlParams.toString();
+	}
+
+	return returnPath;
+}
```

Comment:
Can I suggest this implementation?

```suggestion
function getRelativeUrlWithParameters( queryArgs: Record< string, string > ): string {
	const url = new URL( window.location.href );

	for ( const [ key, value ] of Object.entries( queryArgs ) ) {
		if ( '' !== value ) {
			url.searchParams.set( key, value );
		} else {
			url.searchParams.delete( key );
		}
	}

	return url.pathname + url.search + url.hash;
}
```

Reply:
Great suggestion, [improved it here](https://github.com/Automattic/wp-calypso/pull/63825/commits/dead29ba009b75629fc967d5508bc995024289ce)